src/main/java/Wolox/training/controllers/BookController.java

```diff
@@ -41,78 +55,111 @@ public Iterable findAll() {
     }
 
     @GetMapping("/view/{id}")
-    public Book findById(@PathVariable int id) {
-        return bookRepository.findById(id).orElseThrow(() -> new RuntimeException("The book does not exist"));
+    public Book findById(@PathVariable int id) throws BookDoesNotExistException {
+        return bookRepository.findById(id).orElseThrow(() -> new BookDoesNotExistException("The book does not exist"));
+    }
+
+    @ResponseStatus(HttpStatus.OK)
+    @GetMapping
+    public Book findLocalByIsbn(String isbn) throws BookDoesNotExistException {
```

Comment:
if it's not a method that one could access from outside the application and it's only use for internal purposes you should't add the get mapping and response anotations.